src/services/getFrameKmsService.ts

```diff
@@ -214,17 +243,36 @@ async function processFrameKms() {
                 await fkmStorage.setItem(framekm, JSON.stringify(nextState));
             } else if (status === LteFkmStatus.COMMIT_SUCCESS) {
                 console.log(`Done ${framekm}`);
-                fs.unlinkSync(path.join(FRAMEKM_ROOT_FOLDER, framekm));
-                fs.unlinkSync(path.join(METADATA_ROOT_FOLDER, framekm) + '.json');
+                await unlinkIfExists(path.join(FRAMEKM_ROOT_FOLDER, framekm));
+                await unlinkIfExists(path.join(METADATA_ROOT_FOLDER, framekm) + '.json');
                 await fkmStorage.del(framekm);
             }
             numFrameKmErrorsInARow = 0;
         } catch (e) {
-            console.log('Frame ', framekm, ' failed with error ', e);
+            console.log('Frame ', framekm, 'failed');
             // FetchErrors indicate something is wrong with the network, so we should stop processing
             // and try to recover LTE in the main loop
             if (e instanceof FetchError) {
+                console.log('FetchError', e.message);
                 throw e;
+            } else if (e instanceof FrameKmRequestError) {
+                console.log('FrameKmRequestError', e);
+                console.log(e.errorString);
+                const modifiedErrorString = e.errorString.replace('Error: ', '');
+                if (requestRejectionMessages.includes(modifiedErrorString)) {
+                    console.log('Rejecting framekm', framekm, 'for reason', e.errorString);
+                    await unlinkIfExists(path.join(FRAMEKM_ROOT_FOLDER, framekm));
+                    await unlinkIfExists(path.join(METADATA_ROOT_FOLDER, framekm) + '.json');
+                    await fkmStorage.del(framekm);
+                }
+            } else if (e instanceof FrameKmUploadError) {
+                console.log('FrameKmUploadError', e);
+                if (e.status === 400) {
+                    console.log('Token expired, resetting framekm entry to initial state', framekm);
```

Comment:
I would put ... "token possibly expired..."
as we dont know, in case you forget to refactor to see the actual error in depth

Reply:
or leave a comment to explain the assumption